Dockerfile

```diff
@@ -8,61 +8,43 @@ RUN apt-get update && \
     apt-get clean && rm -rf /var/lib/apt/lists/*
 
-# download extensions as in bundle
-COPY download-extension.sh .
-RUN bash download-extension.sh OAuth;\
-bash download-extension.sh Elastica;\
-bash download-extension.sh CirrusSearch;\
-bash download-extension.sh WikibaseCirrusSearch;\
-bash download-extension.sh UniversalLanguageSelector;\
-bash download-extension.sh cldr;\
-bash download-extension.sh EntitySchema;\
-bash download-extension.sh Babel;\
-bash download-extension.sh ConfirmEdit;\
-bash download-extension.sh Scribunto;\
-bash download-extension.sh VisualEditor;\
-bash download-extension.sh WikibaseManifest;\
-bash download-extension.sh Wikibase; \
-tar xzf OAuth.tar.gz;\
-tar xzf Elastica.tar.gz;\
-tar xzf CirrusSearch.tar.gz;\
-tar xzf WikibaseCirrusSearch.tar.gz;\
-tar xzf UniversalLanguageSelector.tar.gz;\
-tar xzf cldr.tar.gz;\
-tar xzf EntitySchema.tar.gz;\
-tar xzf Babel.tar.gz;\
-tar xzf ConfirmEdit.tar.gz;\
-tar xzf Scribunto.tar.gz;\
-tar xzf VisualEditor.tar.gz;\
-tar xzf WikibaseManifest.tar.gz;\
-tar xzf Wikibase.tar.gz;\
-rm ./*.tar.gz
-
-RUN git clone https://github.com/ProfessionalWiki/WikibaseLocalMedia.git -b REL1_35 WikibaseLocalMedia
-RUN rm -rf WikibaseLocalMedia/.git
-
-# c1233da is the last revision that works with the docker bundle based on mediawiki 1.35
-RUN git clone https://github.com/Wikidata/WikibaseImport.git WikibaseImport && \
-cd WikibaseImport && git checkout c1233da6b7122c55c95d9d925b8f4162de8807e7 && \
+# clone extensions from github, using specific branch
+
+ENV BRANCH=REL1_35
+
+COPY clone-extension.sh .
+
+RUN bash clone-extension.sh OAuth ${BRANCH};\
+bash clone-extension.sh Elastica ${BRANCH};\
+bash clone-extension.sh CirrusSearch ${BRANCH};\
+bash clone-extension.sh WikibaseCirrusSearch ${BRANCH};\
+bash clone-extension.sh UniversalLanguageSelector ${BRANCH};\
+bash clone-extension.sh cldr ${BRANCH};\
+bash clone-extension.sh EntitySchema ${BRANCH};\
+bash clone-extension.sh Babel ${BRANCH};\
+bash clone-extension.sh ConfirmEdit ${BRANCH};\
+bash clone-extension.sh Scribunto ${BRANCH};\
+bash clone-extension.sh VisualEditor ${BRANCH};\
+bash clone-extension.sh WikibaseManifest ${BRANCH};\
+bash clone-extension.sh Wikibase ${BRANCH};\
+bash clone-extension.sh TemplateStyles ${BRANCH};\
+bash clone-extension.sh JsonConfig ${BRANCH};\
+bash clone-extension.sh Lockdown ${BRANCH};\
+bash clone-extension.sh Nuke ${BRANCH};\
+bash clone-extension.sh Math ${BRANCH};\
+bash clone-extension.sh MathSearch ${BRANCH};
+
+# clone extensions not officially distributed by mediawiki
+RUN git clone https://github.com/ProfessionalWiki/WikibaseLocalMedia.git -b ${BRANCH} WikibaseLocalMedia &&\
+rm -rf WikibaseLocalMedia/.git
+
+RUN git clone https://github.com/ciencia/mediawiki-extensions-TwitterWidget.git TwitterWidget &&\
+rm -rf TwitterWidget/.git
+
+# clone WikibaseImport from the mardi fork (no branch needed here, as extension is custom made for the portal)
```

Comment:
If we plan a change from one version to another, it makes sense to have different branches for some time.

Reply:
TwitterWidget has only the main branch.